### ISSUE TITLE
A complete example for nested collections?

### ISSUE THREAD

**Issue by AnatoliiStepaniuk** [outworkers/phantom#678]
Hello! Could you please suggest adding a complete example of nested collections into Phantom documentation and/or source code? I run into several issues trying to model Map[Int, List[Double]] with `Phantom 2.7.6`, as described bellow.
When I try a straight-forward model like this:
```
abstract class NotRatedItemsModel extends CassandraTable[ConcreteNotRatedItemsModel, NotRatedItem] {
  override def tableName: String = "not_rated_items"
  object userId extends IntColumn(this) with PartitionKey
  object items extends MapColumn[Int, List[Double]](this)
  override def fromRow(r: Row): NotRatedItem = NotRatedItem(userId(r), items(r))

abstract class ConcreteNotRatedItemsModel extends NotRatedItemsModel with RootConnector {
  def getItemNotRatedByUser(userId: Int): Future[List[Map[Int, List[Double]]]] = {
    select(_.items)
      .consistencyLevel_=(ConsistencyLevel.ONE)
      .where(_.userId eqs userId)
      .fetch
  }
}
```
It compiles fine, but calling the `getItemNotRatedByUser` method results in error 
`CodecNotFoundException: Codec not found for requested operation: [list<double> <-> scala.collection.immutable.List])`
If I change Scala List to `java.util.List` I get a compile error:
`Cannot find primitive implementation for trait List  object items extends MapColumn[Int, java.util.List[Double]](this)`

Could you please setup a working example for nested collections? Probably adding TypeCodec's for Scala collections would solve the issue and Phantom users would benefit from it!
Thanks a lot!

**Comment by alexflav23** [comment]
Hi @AnatoliiStepaniuk Wait for #668, which should land very shortly, we will no longer use `CodecRegistry` at all,`TypeCodec` is an abomination.

**Comment by alexflav23** [comment]
Hi @AnatoliiStepaniuk Fixed in 2.8.0, going live in the next few hours!!
